error.go

```diff
@@ -27,12 +27,85 @@ package temporal
 import (
 	"errors"
 
+	commonpb "go.temporal.io/temporal-proto/common"
 	"go.temporal.io/temporal-proto/serviceerror"
 
 	"go.temporal.io/temporal/internal"
 	"go.temporal.io/temporal/workflow"
 )
 
+/*
+If activity fails then *ActivityTaskError is returned to the workflow code. The error has important information about activity
+and actual error which caused activity failure. This internal error can be unwrapped using errors.Unwrap() or checked using errors.As().
+Below are the possible types of internal error:
+1) *ApplicationError: (this should be the most common one)
+	*ApplicationError can be returned in two cases:
+		- If activity implementation returns *ApplicationError by using NewApplicationError() API.
+		  The err would contain a message, details, and NonRetryable flag. Workflow code could check this flag and details to determine
+		  what kind of error it was and take actions based on it. The details is encoded payload which workflow code could extract
+		  to strong typed variable. Workflow code needs to know what the types of the encoded details are before extracting them.
+		- If activity implementation returns errors other than from NewApplicationError() API. In this case GetOriginalType()
+		  will return orginal type of an error represented as string. Workflow code could check this type to determine what kind of error it was
+		  and take actions based on the type. These errors are retryable by default, unless error type is specified in retry policy.
+2) *CanceledError:
+	If activity was canceled, internal error will be an instance of *CanceledError. When activity cancels itself by
+	returning NewCancelError() it would supply optional details which could be extracted by workflow code.
+3) *TimeoutError:
+	If activity was timed out (several timeout types), internal error will be an instance of *TimeoutError. The err contains
+	details about what type of timeout it was.
+4) *PanicError:
+	If activity code panic while executing, temporal activity worker will report it as activity failure to temporal server.
+	The SDK will present that failure as *PanicError. The err contains a string	representation of the panic message and
+	the call stack when panic was happen.
+
+Workflow code could handle errors based on different types of error. Below is sample code of how error handling looks like.
+
+_, err := workflow.ExecuteActivity(ctx, MyActivity, ...).Get(nil)
+if err != nil {
+	var applicationErr *ApplicationError
+	if errors.As(err, &applicationError) {
+		// handle activity errors (created via NewApplicationError() API)
+		if !applicationErr.NonRetryable() {
+			// manually retry activity
```

Comment:
I'm not sure why anyone would ever check this flag as this is mostly for the service to deal with it.